src/components/Api.js

```diff
@@ -0,0 +1,146 @@
+class Api {
+  constructor({ baseUrl, authToken, headers }) {
+    this._baseUrl = baseUrl;
+    this._authToken = authToken;
+    this._headers = headers;
+  }
+
+  _processResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    })
+      .then((res) =>
+        res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)
+      )
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very end of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.